var/exp_repo/experiments/caliper/experiment.py

```
@@ -0,0 +1,94 @@
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+from benchpark.directives import variant
+from benchpark.experiment import Experiment
+
+
+class Caliper(Experiment):
+    variant(
+        "caliper",
+        default="none",
+        values=(
+            "none",
+            "time",
+            "mpi",
+            "cuda",
+            "topdown-counters-all",
+            "topdown-counters-toplevel",
+            "topdown-all",
+            "topdown-toplevel",
+        ),
+        multi=True,
+        description="caliper mode",
+    )
+
+    def is_enabled(self):
+        return not self.spec.satisfies("caliper=none")
+
+    def compute_modifiers_section(self):
+        modifier_list = []
+        if not self.spec.satisfies("caliper=none"):
+            for var in list(self.spec.variants["caliper"]):
+                if var != "time":
+                    caliper_modifier_modes = {}
+                    caliper_modifier_modes["name"] = "caliper"
+                    caliper_modifier_modes["mode"] = var
+                    modifier_list.append(caliper_modifier_modes)
+            # Add time as the last mode
+            modifier_list.append({"name": "caliper", "mode": "time"})
+        return modifier_list
+
+    def compute_spack_section(self):
+        # set package versions
+        caliper_version = "master"
+
+        # get system config options
+        # TODO: Get compiler/mpi/package handles directly from system.py
+        system_specs = {}
+        system_specs["compiler"] = "default-compiler"
+        if self.spec.satisfies("programming_model=cuda"):
+            system_specs["cuda_arch"] = "{cuda_arch}"
+        if self.spec.satisfies("programming_model=rocm"):
+            system_specs["rocm_arch"] = "{rocm_arch}"
+
+        # set package spack specs
+        package_specs = {}
+
+        if not self.spec.satisfies("caliper=none"):
```

I get the impression that some combinations of `caliper=...` are not allowed. I don't know if we support `conflict` yet, but FWIW you can raise an exception here for forbidden combinations.